commit cffbb336ff4a19b2c70aa728535d2deac98fb7d4
Author: Radek Doulik <rodo@novell.com>
Date:   Wed Jan 5 22:43:00 2011 +0100

    fix shapes rendering order n#656934
    
     - it was broken for master pages containing group shapes

diff --git slideshow/source/engine/slide/slideimpl.cxx slideshow/source/engine/slide/slideimpl.cxx
index 43eb012..95c047b 100644
--- slideshow/source/engine/slide/slideimpl.cxx
+++ slideshow/source/engine/slide/slideimpl.cxx
@@ -1140,7 +1140,7 @@ bool SlideImpl::loadShapes()
                         mpLayerManager->addShape( rShape );
                 }
                 
-                nCurrCount = xMasterPageShapes->getCount() + 1;
+                nCurrCount = aMPShapesFunctor.getImportedShapesCount();
             }
             catch( uno::RuntimeException& )
             {
diff --git slideshow/source/inc/shapeimporter.hxx slideshow/source/inc/shapeimporter.hxx
index 0e7023b..384f35b 100644
--- slideshow/source/inc/shapeimporter.hxx
+++ slideshow/source/inc/shapeimporter.hxx
@@ -103,6 +103,7 @@ public:
     */
     bool isImportDone() const;
 	 
+    double getImportedShapesCount() { return mnAscendingPrio; }
 private:
     bool isSkip( ::com::sun::star::uno::Reference<
                  ::com::sun::star::beans::XPropertySet> const& xPropSet,
